src/main/java/spring/config/AppConfig.java

```diff
@@ -19,7 +19,7 @@
         "spring.dao"
 })
 public class AppConfig {
-    private Environment environment;
+    private final Environment environment;
```

Comment:
why do you use "final" here, but don't use it with UserService and UserDao, let's keep your code consistent